### PR TITLE
Algorithm datasheet for SIDH-CLN16.

### LINKED ISSUE #163: Algorithm datasheet for Picnic and SIDH
Christian, I would like to have "algorithm datasheets" providing a bit more information about each algorithm in liboqs (e.g., https://github.com/open-quantum-safe/liboqs/blob/master/docs/Algorithm%20data%20sheets/kex_lwe_frodo.md).  Can I ask you to make up something similar for Picnic?  It doesn't have to match exactly, use your judgement on what information is appropriate for a signature scheme.

### PR DESCRIPTION
**Pull request by dconnolly**
Fixes #163 

@christianpaquin @dstebila I took a swing at this. Not sure about the `scan_build` part, haven't tested that yet. Also I included the uncompressed/compressed key sizes, even though it appears the kex is configured to be uncompressed by default. I vaguely described the underlying problem in the top level description as just 'the hidden isogeny' problem, I can pin that to the SSDDH problem as below if that's better.

### CODE REVIEW

File: docs/Algorithm data sheets/kex_sidh_cln16.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+- [CLN16] Craig Costello, Patrick Longa, and Michael Naehrig. Efficient algorithms for supersingular isogeny
+Diffie-Hellman. In *CRYPTO 2016*, volume 9814 of LNCS, pp 572-601. Springer, 2016. [https://eprint.iacr.org/2016/413](https://eprint.iacr.org/2016/413)
+- [DJP11] Luca De Feo and David Jao and Jérôme Plût. Towards quantum-resistant cryptosystems from supersingular elliptic curve isogenies. In *PQCRYPTO 2011*, volume 7071 of LNCS, pages 19-34. Springer, 2011. [https://eprint.iacr.org/2011/506](https://eprint.iacr.org/2011/506)
+
+
+Security
+--------
+
+**Security model:** Authenticated-links adversarial model
```
Comment (dstebila) [inline docs/Algorithm data sheets/kex_sidh_cln16.md]: For the datasheets for Frodo and BCNS15, we write "Unauthenticated key exchange / passive (IND-CPA) key encapsulation mechanism".  I think that's an accurate description here too.  Would that be okay, @dconnolly?

Comment (dconnolly) [inline docs/Algorithm data sheets/kex_sidh_cln16.md]: I think that's fair, I included this model as it is proven in the DJP11 paper, but I missed a relevant conclusion from CLN16: `the Canneti-Krawczyk model assumes perfectly authenticated links which effectively forces adversaries to be passive eavesdroppers`. I'll update this. :)

**Comment by christianpaquin** [comment]
Taking the post-defcon week off. I'll take a look on Monday!

On Aug 4, 2017, at 12:43 PM, Deirdre Connolly <notifications@github.com<mailto:notifications@github.com>> wrote:


@dconnolly commented on this pull request.

________________________________

In docs/Algorithm data sheets/kex_sidh_cln16.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fopen-quantum-safe%2Fliboqs%2Fpull%2F165%23discussion_r131433638&data=02%7C01%7Ccpaquin%40microsoft.com%7Cfb589c6dfb6e4aa8e02c08d4db57f187%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636374618140449252&sdata=6JR7YzVJ0OUvEnfkheCw3Z3%2FLNAV7X4QwUuCAoHQajw%3D&reserved=0>:

> +
+**Name:** SIDH-CLN16
+
+**Description:** Key exchange protocol proposed by Costello et al. [CLN16] based on the hidden supersingular isogeny problem. Instantiation of the SIDH scheme proposed by De Feo, Jao and Plut in [DJP11].
+
+**Supporting research:**
+
+- [CLN16] Craig Costello, Patrick Longa, and Michael Naehrig. Efficient algorithms for supersingular isogeny
+Diffie-Hellman. In *CRYPTO 2016*, volume 9814 of LNCS, pp 572-601. Springer, 2016. [https://eprint.iacr.org/2016/413](https://eprint.iacr.org/2016/413)
+- [DJP11] Luca De Feo and David Jao and Jérôme Plût. Towards quantum-resistant cryptosystems from supersingular elliptic curve isogenies. In *PQCRYPTO 2011*, volume 7071 of LNCS, pages 19-34. Springer, 2011. [https://eprint.iacr.org/2011/506](https://eprint.iacr.org/2011/506)
+
+
+Security
+--------
+
+**Security model:** Authenticated-links adversarial model


I think that's fair, I included this model as it is proven in the DJP11 paper, but I missed a relevant conclusion from CLN16: the Canneti-Krawczyk model assumes perfectly authenticated links which effectively forces adversaries to be passive eavesdroppers. I'll update this. :)

**Comment by dstebila** [comment]
Obsoleted by #166.

**Pull request closed without merging**